pkg/dependencies/factory.go

```diff
@@ -636,8 +642,39 @@ func (f *Factory) WithNetworking(clusterConfig *v1alpha1.Cluster) *Factory {
 	return f
 }
 
+func (f *Factory) WithCNIInstaller(spec *cluster.Spec, provider providers.Provider) *Factory {
+	if spec.Cluster.Spec.ClusterNetwork.CNIConfig.Kindnetd != nil {
+		f.WithKubectl()
+	} else {
+		f.WithKubectl().WithCiliumTemplater()
+	}
+
+	f.buildSteps = append(f.buildSteps, func(ctx context.Context) error {
+		if f.dependencies.CNIInstaller != nil {
+			return nil
+		}
+
+		if spec.Cluster.Spec.ClusterNetwork.CNIConfig.Kindnetd != nil {
+			f.dependencies.CNIInstaller = kindnetd.NewInstallerForSpec(f.dependencies.Kubectl, spec)
+		} else {
+			f.dependencies.CNIInstaller = cilium.NewInstallerForSpec(
+				cilium.NewRetrier(f.dependencies.Kubectl),
+				f.dependencies.CiliumTemplater,
+				cilium.Config{
+					Spec:              spec,
+					AllowedNamespaces: maps.Keys(provider.GetDeployments()),
+				},
+			)
+		}
+
+		return nil
+	})
+
+	return f
+}
+
 func (f *Factory) WithCiliumTemplater() *Factory {
-	f.WithHelm()
+	f.WithHelm(executables.WithInsecure())
```

Comment:
I'm not sure what `f.WithHelm(executables.WithInsecure())` does, but it was being used before to build the the helm passed to `Cilium`